Lab5/src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,104 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class GrassField extends WorldMap{
+
+    private int numberOfGrass;
+    private List<Grass> grassList;
+
+    public GrassField(int numberOfGrass) {
+        this.grassList = new ArrayList<Grass>();
+        this.animalList = new ArrayList<Animal>();
+        this.numberOfGrass = numberOfGrass;
+        for (int i=1; i <= numberOfGrass;i++ ){
+            createGrass();
+            }
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        boolean canMove = super.canMoveTo(position);
+        if (canMove) {
+            // if animal steps on the grass, the grass vanishes and appears in new free place
+            if (this.isOccupiedByGrass(position)) {
+                this.grassList.removeIf(grass -> grass.isAt(position));
+                createGrass();
+            }
+        }
+        return canMove;
+    }
+
+    @Override
+    public boolean isOccupied(Vector2d position) {
+        boolean isOccupied = super.isOccupied(position);
+        if (this.objectAt(position) != null) {
```

Comment:
hm, ale to dokładnie robi bazowa implementacja - moim zdaniem w ogóle nadpisywanie `isOccupied()` jest niepotrzebne.